odoo/addons/base/models/ir_asset.py

```diff
@@ -44,8 +44,49 @@ def is_wildcard_glob(path):
 
 
 def _glob_static_file(pattern):
-    files = glob(pattern, recursive=True)
-    return sorted((file, os.path.getmtime(file)) for file in files if file.rsplit('.', 1)[-1] in ASSET_EXTENSIONS)
+    # simplified glob support (fnmatch is not precise enough)
+    # the purpose of this algo is to be as fast as possible.
+    # faster than pathlib.glob, glob.glob, other_obscure_glob,...
+    cache = service.server._static_file_cache
+    assert pattern[0] == '/'
+    parts = pattern.split('/')[1:]
+    nb_parts = len(parts)
+    stack = [(cache, 0, [''])]
+    results = []
+
+    while stack:
+        cache, index, file_path = stack.pop()
+        if isinstance(cache, float):
+            if index == nb_parts:
+                results.append((os.sep.join(file_path), cache))
+        elif index != nb_parts:
+            part = parts[index]
+            if part == "*":
+                for directory, directory_cache in cache.items():
+                    stack.append((directory_cache, index + 1, file_path + [directory]))
+            elif part == "**":
+                stack.append((cache, index + 1, file_path))  # ** can match empty
+                for directory, directory_cache in cache.items():
+                    stack.append((directory_cache, index, file_path + [directory]))  # ** can match multiple directory
+                    if index + 1 == nb_parts:
+                        stack.append((directory_cache, index + 1, file_path + [directory]))  # can match a file
```

Comment:
This doesn't look necessary to me.  Combining the line above with two extra loops should have the same effect.